platform-sdk/docs/proposals/TSS-Block-Signing/tss-block-signing.md

```diff
@@ -0,0 +1,286 @@
+`shares`, whose signatures on a message can be aggregated to produce a `ledger signature` equivalent to the signature
+generated by the ledger's private key signing the same message. Ledger signatures are verifiable by the ledger id. The
+value of our chosen TSS scheme is that a network of nodes is able to transfer the ability to generate ledger
+signatures to another network of nodes without revealing the ledger private key.
+
+TSS Proposals:
+
+1. The `TSS-Library` proposal contains the cryptographic primitives and algorithms needed to implement TSS.
+2. The `TSS-Roster` proposal introduces the data structure of a consensus `Roster` to the platform.
+3. The `TSS-Ledger-ID` proposal introduces a ledger id for existing networks.
+4. This proposal (`TSS-Block-Signing`) introduces the methodology for signing blocks.
+5. The `TSS-Ledger-ID-Updates` proposal covers the process of resetting and transplanting ledger ids between networks.
+6. The `TSS-Ledger-New-Networks` proposal covers the process of setting up a new network with a ledger id.
+
+This proposal, `TSS-Block-Signing` covers the following:
+
+- `TssBaseService` API
+  - Add a method for requesting ledger signatures on a message.
+  - Add a method for registering consumers of ledger signatures.
+- `TssBaseService` Implementation
+  - Each node gossips a `TssShareSignatureTransaction` for each share signature it produces.
+  - A threshold number of `TssShareSignatureTransaction` are aggregated to produce the ledger signature.
+
+### Goals
+
+The following are goals of this proposal:
+
+1. That existing networks are able to create ledger signatures if the active roster has key material that can
+   recover the network's ledger id.  (See `TSS-Ledger-ID`)
+
+### Non-Goals
+
+The following is not part of the scope of this proposal:
+
+1. The ability to create ledger signatures for new networks.
+2. Integrating this capability into the broader block-stream effort.
+
+### Dependencies, Interactions, and Implications
+
+Dependencies on `TSS-Library`:
+
+- Upon availability of the implementation:
+  - This proposal can be tested with a pre-rendered set of shares, ledger private key, and ledger id.
+
+Dependencies on `TSS-Ledger-ID`:
+
+- Upon availability of the API:
+  - The `TssBaseService` API can be extended by this proposal.
+- Upon availability of the implementation:
+  - This proposal's full behavior can be tested in integration tests.
+
+Impacts to the Services Team:
+
+- The ability to sign blocks with TSS is blocked until this proposal delivers its capability.
+
+Implication Of Completion:
+
+- Signing blocks and construction of block proofs with TSS signatures is unblocked.
+
+### Requirements
+
+The core requirements governing the TSS proposals are stated in `TSS-Ledger-ID`.
+
+Blocks Signing Specific Requirements:
+
+1. A block is considered signed if its root hash has a ledger signature or a following block's root hash has a
+   ledger signature.
```

Comment:
Why not simplify and only require a signature on the following block at all times?

Conversely, why not harden the requirement and require a signature on the very block that is to be considered signed at all times?

If both the options are viable, then this needs to be explained further and in greater details as to why, and under what conditions the options are used.

Reply:
Thanks for the extra details and wording in the doc. That helps.